content/client-lib-development-guide/features.textile

```diff
@@ -300,6 +300,8 @@ h3(#rest-channel). Channel
 *** @(RSL6a2)@ A set of tests must exist to ensure that the client library provides interoperability for the @extras@ field which is a JSON-encodable object (ie a value that represents a JSON @object@ value and supports serialization to and from JSON text). The test, at a minimum, should publish a message with an @extras@ object such as @{"push":[{"title":"Testing"}]}@ and ensure it is received with an equivalent JSON-encodable object
 ** @(RSL6b)@ If, for example, incompatible encryption details are provided or invalid Base64 is detected in the message payload, an error message will be sent to the logger, but the message will still be delivered with last successful decoding and the @encoding@ field. For example, if a message had a decoding of "utf-8/cipher+aes-128-cbc/base64", and the payload was successfully Base64 decoded but the payload could not be decrypted because the @CipherParam@ details were not configured, the message would be delivered with a binary payload and an @encoding@ with the value "utf-8/cipher+aes-128-cbc". Additional steps need to be taken if decoding failed on "vcdiff" encoding; see "RTL18":#RTL18
 * @(RSL7)@ @Channel#setOptions@ takes a @ChannelOptions@ object and sets or updates the stored channel options, then indicates success
+* @(RSL8)@ @Channel#status@ function: makes a http get request to @rest.ably.io/channels/<channelId>@
+** @(RSL8a)@ Returns a @ChannelDetails@ object or an @ErrorInfo@ object with details of the error that has occurred
```

Comment:
I would expect that in most languages the `ErrorInfo` would be thrown rather than returned. I'd suggest either:
1. Change this to "Returns a @ChannelDetails@ object or results in an @ErrorInfo@ object with details of the error that has occurred"
2. Remove this spec point and consider it already covered by `TI2`

Reply:
So this is was written from my point of view as a Go developer working in a language where it's idiomatic to communicate errors via an explicit, separate return values. In my Go world errors are not thrown, they are always returned and can be nil. 

I think this certainly proves it's exceptionally hard to write a specification that makes sense in multiple languages. So I've gone with suggestion 2 and removed the information about `ErrorInfo`.